pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -233,28 +288,46 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 					KubernetesServiceName:      svc.Attributes.Name,
 				},
 			}
-			tcpChain := &listener.FilterChain{
-				Filters: lb.buildInboundNetworkFilters(cc),
-				Name:    cc.clusterName,
-			}
+			var tcpChain, httpChain *listener.FilterChain
+			origDst := svc.GetAddressForProxy(lb.node) + ":" + portString
 			httpClusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP, "http", svc.Hostname, port.Port)
-			cc.clusterName = httpClusterName
-			httpChain := &listener.FilterChain{
-				Filters: lb.buildWaypointInboundHTTPFilters(svc, cc),
-				Name:    cc.clusterName,
+			if len(svcAddresses) > 0 {
```

Comment:
feature flag should cover this too I think?

Reply:
Ack - added